11.1.0 Changelog (#4859)
cypress-io/cypress-documentation: content/_changelogs/11.1.0.md

## 11.1.0

_Released 11/14/2022_

**Features:**

- Next.js v13 is now supported.
  [#24396](https://github.com/cypress-io/cypress/pull/24396)

**Bugfixes:**

> - Fixed an 11.0.0 regression where users couldn't migrate to 11 without going

Reviewer: ```suggestion
- Fixed an 11.0.0 regression where Typescript projects couldn't migrate to 11 without going
```

> - Fixed an 11.0.0 regression where Typescript projects couldn't migrate to 11 without going
>   through 10 first. Fixes

Reviewer 2: Won't all project always need to be migrated to 10 before the 11.0 migration occurs?
```suggestion
- Fixed an 11.0.0 regression where the Migration flow would error and hang for Typescript projects. Fixes
```

Author: Not if they're already on 10? But I can probably clarify this.

Reviewer: Cypress 11 migration is the same as Cypress 10 migration, it's based on whether we detect a `cypress.json` file. People are able to migrate directly to 11 from 9 or lower.

  [#24643](https://github.com/cypress-io/cypress/issues/24643)

> - Fixed an 11.0.0 regression where `cypress run` crashes when using the junit

Reviewer 2: ```suggestion
- Fixed an 11.0.0 regression where `cypress run` crashed when using the junit
```

  reporter and an assertion fails. Fixes
  [#24652](https://github.com/cypress-io/cypress/issues/24652)
- Fixed Typescript types for `testIsolation`. Fixes
  [#23911](https://github.com/cypress-io/cypress/issues/23911)

> - Sessions will no longer be stored if validation fails. Fixes

Reviewer 2: This prevents a previously failed session from being restored in the next test.

  [#24208](https://github.com/cypress-io/cypress/issues/24208)
- Updates were made to enhance the cy.session() recovered error experience. With
  this change, cy.session() will no longer fail validation when false is
  returned from the validate callback. Fixes
  [#21398](https://github.com/cypress-io/cypress/issues/21398),
  [#24205](https://github.com/cypress-io/cypress/issues/24205), and

>   [#24593](https://github.com/cypress-io/cypress/issues/24593).

Reviewer 2: We had chatted about this on slack. This is more of an enhancement vs bug fix, but this should prob be called out under `Experimental Breaking Changes:` with the change to return false

> - `mount()` now accept a extensions.directives option, which will be registered

Reviewer 3: ```suggestion
- `mount()` now accepts an `extensions.directives` option, which will be registered
```

> - `mount()` now accepts an `extensions.directives` option, which will be registered

Reviewer 2: This sounds like a feature. If this is a bug fix, what does this resolve?

## 11.1.0

_Released 11/14/2022_

**Features:**

- Next.js v13 is now supported.
  [#24396](https://github.com/cypress-io/cypress/pull/24396)
- `mount()` now accepts an extensions.directives option, which will be registered
  globally for Vue2 component testing.
  [#24488](https://github.com/cypress-io/cypress/pull/24488)

**Bugfixes:**

- Fixed an 11.0.0 regression where the migration workflow would error and hang
  for TypeScript projects. Fixes
  [#24643](https://github.com/cypress-io/cypress/issues/24643)
- Fixed an 11.0.0 regression where `cypress run` crashed when using the junit

>   reporter and an assertion fails. Fixes

Author: ```suggestion
  reporter and an assertion failed. Fixes
```

  [#24652](https://github.com/cypress-io/cypress/issues/24652)

> - Fixed Typescript types for `testIsolation`. Fixes

Author: ```suggestion
- Fixed TypeScript types for `testIsolation`. Fixes
```

  [#23911](https://github.com/cypress-io/cypress/issues/23911)

> - Sessions will no longer be stored if validation fails preventing a previously

Author: ```suggestion
- Sessions will no longer be stored if validation fails which will prevent a previously
```
